src/treeViewLemma.ts

```diff
@@ -71,32 +63,54 @@ export class Entry extends vscode.TreeItem {
     super(label, collapsibleState);
     this.tooltip = `${this.label}`;
   }
-  public static async initialize(context: vscode.ExtensionContext) {
-    Entry.rootPath = context.extensionPath;
-    const fs = vscode.workspace.fs;
-    const hebrewDataPath = vscode.Uri.file(
-      path.join(Entry.rootPath, "data", "local", "treeViewHebrewWordData.json")
-    );
-    const greekDataPath = vscode.Uri.file(
-      path.join(Entry.rootPath, "data", "local", "treeViewGreekWordData.json")
-    );
-    console.log("hebrewDataPath:", hebrewDataPath);
-    console.log("greekDataPath:", greekDataPath);
 
-    await this.readAndProcessFile(hebrewDataPath, "hebrew");
-    await this.readAndProcessFile(greekDataPath, "greek");
-  }
+  public static async initialize(): Promise<void> {
+    // Need to reset Entry because this is called if the workspace folder changes
+    Entry.entriesMap.clear();
+    Entry.rootPath = "";
 
+    if (
+      vscode.workspace.workspaceFolders &&
+      vscode.workspace.workspaceFolders.length > 0
+    ) {
+      const workspaceFolder = vscode.workspace.workspaceFolders[0].uri.fsPath;
+
+      const hebrewPath = path.join(
+        workspaceFolder,
+        `${languagePath}${languageCode}/hebrew`
+      );
+      const greekPath = path.join(
+        workspaceFolder,
+        `${languagePath}${languageCode}/greek`
+      );
+
+      const hebrewExists = fs.existsSync(hebrewPath);
+      const greekExists = fs.existsSync(greekPath);
+
+      if (hebrewExists && greekExists) {
+        return new Promise((resolve) => {
+          setTimeout(() => {
```

Comment:
What is the purpose of the promise and timeout? There's nothing async in here

Reply:
This is to resolve a timing issue that I had whereby the tree view did not populate with data.

In previous version of `extension.ts` I had:
``` 
// Register the TreeDataProvider
  const lemmaTreeDataProvider = new LemmaTreeDataProvider(context);
  vscode.window.registerTreeDataProvider(
    "treeView-lemma",
    lemmaTreeDataProvider
  );
```
and in `LemmaTreeDataProvider.getChildren()` in `treeViewLemma.ts` I had: 
```
   if (!element) {
      // Return root level entries (level 1)
      // Commenting this out as there is a timing issue in that when this is called, the Entry.entriesMap is not yet populated
      // const rootEntries = Array.from(Entry.entriesMap.values()).filter(
      //   (entry) => entry.key.startsWith("1")
      // );

      // This is a temporary solution to return root entries
      const rootEntries = new Array<Entry>();
      rootEntries[0] = new Entry(
        1,
        "1H001",
        "Hebrew",
        vscode.TreeItemCollapsibleState.Collapsed
      );
      rootEntries[1] = new Entry(
        1,
        "1G002",
        "Greek",
        vscode.TreeItemCollapsibleState.Collapsed
      );
```

As mentioned in comment above, when the first call of `LemmaTreeDataProvider.getTreeItem` was performed to populate 1st level of tree view, the `Entry.entriesMap` had been yet been populated due to timing issues. As a result the tree was never populated. (By the time the extension made the 2nd call to `getChildren()` the `Entry.entriesMap` was populated but the 1st level was empty so no data was added to the tree view

To ensure that the initialization of `Entry.entriesMap` completed before `getTreeItem` was called the 1st time, I rewrote as follows:

Changed `extension.ts` and added an `await`
```
const lemmaTreeDataProvider = new LemmaTreeDataProvider();
  await lemmaTreeDataProvider.initialize(); // Wait for initialization
  const lemmaTreeView = vscode.window.createTreeView("lexicon-translation.treeView-lemma", { treeDataProvider: lemmaTreeDataProvider, showCollapseAll: true });
```

To enable the calling of `await` above, modified `treeViewLemma.ts` as follows:

added this to `LemmaTreeDataProvider`:
```
async initialize() {
    await Entry.initialize(); // Ensure Entry is initialized
  }
```
then added this Promise with short timeout:
```
if (hebrewExists && greekExists) {
        return new Promise((resolve) => {
          setTimeout(() => {
            // Setup Entry static variables
            Entry.rootPath = workspaceFolder;
            this.readAndProcessFile(hebrewData, "hebrew");
            this.readAndProcessFile(greekData, "greek");
            resolve();
          }, 1000);
        });
```

Let me know if this makes sense or if there is a better way to resolve my initial problem.